### PR TITLE
Support tensorflow 2.16

### PR DESCRIPTION
**Pull request by uri-granta**
Support and test against TF 2.16.

Note that this also sets fail-fast to false, which makes handling fragile tests easier.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,14 +6,13 @@
 
 requirements = [
     "deprecated",
-    "gpflow>=2.6.3",
-    "numpy",
+    "gpflow>=2.9.2",
+    "numpy<2",
```
Comment (khurram-ghani) [inline setup.py:10]: What is the issue with `numpy>=2`?

File: .github/workflows/quality-check.yaml [hunk 2]
```diff
@@ -20,9 +20,10 @@ jobs:
   check-and-test:
     runs-on: ubuntu-20.04
     strategy:
+      fail-fast: false
```
Comment (khurram-ghani) [inline .github/workflows/quality-check.yaml:23]: Just to check, presumably this wouldn't make testing much slower anyway, as gpflux tests are very fast; correct?

**Review by khurram-ghani — APPROVED** [review]
Looks good. Just a couple of minor questions.

File: setup.py (hunk 1 above)
Comment (uri-granta) [reply setup.py:10]: It's a major, breaking release: e.g. see https://numpy.org/devdocs/numpy_2_0_migration_guide.html.

We'll certainly want to have a conversation about supporting it at some point (possibly company wide), but at this point it probably makes sense to just wait until the dust settles (it was release just a few days ago).

File: .github/workflows/quality-check.yaml (hunk 2 above)
Comment (uri-granta) [reply .github/workflows/quality-check.yaml:23]: The tests take just as long as before if they all pass. However, if one of them fails early on, then previously we would have cancelled all the others, while now we will wait until they all finish (around 10 minutes).

**Pull request merged**